cmd/config-migrate.go

```diff
@@ -32,241 +31,239 @@ import (
-	version, err := GetVersion(getConfigFile())
-	if err != nil {
-		if os.IsNotExist(err) {
-			return nil
-		}
-		return err
-	}
+// DO NOT EDIT following message template, please open a github issue to discuss instead.
+var configMigrateMSGTemplate = "Configuration file migrated from version '%s' to '%s' successfully."
 
-	// Conditional to migrate only relevant config versions.
-	// Upon success migration continues to the next version in sequence.
-	switch version {
+// Migrates all config versions from "2" to latest.
+func migrateConfig(configBytes []byte) ([]byte, error) {
+	var err error
+	switch configGetVersion(configBytes) {
 	case "2":
-		// Migrate version '2' to '3'.
-		if err = migrateV2ToV3(); err != nil {
-			return err
+		configBytes, err = migrateV2ToV3(configBytes)
+		if err != nil {
+			return nil, err
 		}
 		fallthrough
 	case "3":
-		// Migrate version '3' to '4'.
-		if err = migrateV3ToV4(); err != nil {
-			return err
+		configBytes, err = migrateV3ToV4(configBytes)
+		if err != nil {
+			return nil, err
 		}
 		fallthrough
 	case "4":
-		// Migrate version '4' to '5'.
-		if err = migrateV4ToV5(); err != nil {
-			return err
+		configBytes, err = migrateV4ToV5(configBytes)
+		if err != nil {
+			return nil, err
 		}
 		fallthrough
 	case "5":
-		// Migrate version '5' to '6.
-		if err = migrateV5ToV6(); err != nil {
-			return err
+		configBytes, err = migrateV5ToV6(configBytes)
+		if err != nil {
+			return nil, err
 		}
 		fallthrough
 	case "6":
-		// Migrate version '6' to '7'.
-		if err = migrateV6ToV7(); err != nil {
-			return err
+		configBytes, err = migrateV6ToV7(configBytes)
+		if err != nil {
+			return nil, err
 		}
 		fallthrough
 	case "7":
-		// Migrate version '7' to '8'.
-		if err = migrateV7ToV8(); err != nil {
-			return err
+		configBytes, err = migrateV7ToV8(configBytes)
+		if err != nil {
+			return nil, err
 		}
 		fallthrough
 	case "8":
-		// Migrate version '8' to '9'.
-		if err = migrateV8ToV9(); err != nil {
-			return err
+		configBytes, err = migrateV8ToV9(configBytes)
+		if err != nil {
+			return nil, err
 		}
 		fallthrough
 	case "9":
-		// Migrate version '9' to '10'.
-		if err = migrateV9ToV10(); err != nil {
-			return err
+		configBytes, err = migrateV9ToV10(configBytes)
+		if err != nil {
+			return nil, err
 		}
 		fallthrough
 	case "10":
-		// Migrate version '10' to '11'.
-		if err = migrateV10ToV11(); err != nil {
-			return err
+		configBytes, err = migrateV10ToV11(configBytes)
+		if err != nil {
+			return nil, err
 		}
 		fallthrough
 	case "11":
-		// Migrate version '11' to '12'.
-		if err = migrateV11ToV12(); err != nil {
-			return err
+		configBytes, err = migrateV11ToV12(configBytes)
+		if err != nil {
+			return nil, err
 		}
 		fallthrough
 	case "12":
-		// Migrate version '12' to '13'.
-		if err = migrateV12ToV13(); err != nil {
-			return err
+		if err != nil {
+			return nil, err
 		}
+		configBytes, err = migrateV12ToV13(configBytes)
```

Comment:
this needs to be moved before error check

Reply:
good catch